test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientTests.cs

```diff
@@ -35,18 +36,29 @@ public class SqlClientTests : IDisposable
              2) Set OTEL_SQLCONNECTIONSTRING as: Data Source=127.0.0.1,5433; User ID=sa; Password=Pass@word
          */
 
-        private const string SqlConnectionStringEnvVarName = "OTEL_SQLCONNECTIONSTRING";
         private const string TestConnectionString = "Data Source=(localdb)\\MSSQLLocalDB;Database=master";
 
-        private static readonly string SqlConnectionString = SkipUnlessEnvVarFoundTheoryAttribute.GetEnvironmentVariable(SqlConnectionStringEnvVarName);
-
         private readonly FakeSqlClientDiagnosticSource fakeSqlClientDiagnosticSource;
 
+        // Please note this image is not compatible with macOS running on Apple silicon.
+        // Maybe we should replace it with mcr.microsoft.com/azure-sql-edge (the SqlEdge module).
+        private readonly MsSqlContainer mssqlContainer = new MsSqlBuilder().Build();
```

Comment:
If `mcr.microsoft.com/azure-sql-edge` works on Apple silicon, we could use an expression like the following to decide which image to use

```csharp
RuntimeInformation.IsOSPlatform(OSPlatform.OSX) 
&& RuntimeInformation.ProcessArchitecture == Architecture.Arm64
```

Reply:
Right now, it requires a bit more abstraction. The modules do not share a common database base class or interface, but this is not much of a problem. Why not use Azure SQL Edge for all platforms?